### PR TITLE
vscode-extensions.ms-python.python: 2022.11.11881005 -> 2022.15.12711056

### PR DESCRIPTION
**Pull request by linsui**
###### Description of changes
Also add updateScript

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/editors/vscode/extensions/python/default.nix [hunk 1]
```diff
@@ -41,7 +48,34 @@ vscode-utils.buildVscodeMarketplaceExtension {
...
+    if [[ $version != ${mktplcRef.version} ]]; then
+      tmp=$(mktemp)
+      curl -Lo $tmp $(echo ${(import ../mktplcExtRefToFetchArgs.nix mktplcRef).url} | sed "s|${mktplcRef.version}|$version|")
+      hash=$(nix hash file --type sha256 --base32 --sri $tmp)
+      sed -i -e "s|${mktplcRef.sha256}|$hash|" -e "s|${mktplcRef.version}|$version|" pkgs/applications/editors/vscode/extensions/python/default.nix
+    fi
+  '';
+
   meta = with lib; {
+    homepage = "https://marketplace.visualstudio.com/items?itemName=ms-python.python";
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/python/default.nix:78]: Description is missing.

File: pkgs/applications/editors/vscode/extensions/python/default.nix [hunk 2]
```diff
@@ -6,14 +6,21 @@
...
 }:
 
-vscode-utils.buildVscodeMarketplaceExtension {
+vscode-utils.buildVscodeMarketplaceExtension rec {
   mktplcRef = {
     name = "python";
     publisher = "ms-python";
-    version = "2022.11.11881005";
-    sha256 = "sha256-8NH/aWIAwSpVRi3cvBCpvO8MVzIoRaXxADmWp6DuUb8=";
+    version = "2022.15.12631011";
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/python/default.nix:22]:
```suggestion
    version = "2022.15.12711056";
```

File: pkgs/applications/editors/vscode/extensions/python/default.nix [hunk 3]
```diff
@@ -41,7 +48,34 @@ vscode-utils.buildVscodeMarketplaceExtension {
...
+
+    api=$(curl -s 'https://marketplace.visualstudio.com/_apis/public/gallery/extensionquery' \
+      -H 'accept: application/json;api-version=3.0-preview.1' \
+      -H 'content-type: application/json' \
+      --data-raw '{"filters":[{"criteria":[{"filterType":7,"value":"${mktplcRef.publisher}.${mktplcRef.name}"}]}],"flags":512}')
+    version=$(echo $api | sed -n -E 's|^.*"version":"([0-9.]+)".*$|\1|p')
+
+    if [[ $version != ${mktplcRef.version} ]]; then
+      tmp=$(mktemp)
+      curl -Lo $tmp $(echo ${(import ../mktplcExtRefToFetchArgs.nix mktplcRef).url} | sed "s|${mktplcRef.version}|$version|")
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/python/default.nix:71]: Maybe `curl` should be silent here too.

File: pkgs/applications/editors/vscode/extensions/python/default.nix (hunk 1 above)
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/python/default.nix:78]:
```suggestion
    downloadPage = "https://marketplace.visualstudio.com/items?itemName=ms-python.python";
    homepage = "https://github.com/Microsoft/vscode-python";
```

changelog is missing too.

**Review by superherointj — APPROVED** [review]

**Comment by superherointj** [comment]
As you have added an update script, on next update, check for logs at: https://r.ryantm.com/log/

File: pkgs/applications/editors/vscode/extensions/python/default.nix [hunk 4]
```diff
@@ -41,7 +48,37 @@ vscode-utils.buildVscodeMarketplaceExtension {
...
+      hash=$(nix hash file --type sha256 --base32 --sri $tmp)
+      sed -i -e "s|${mktplcRef.sha256}|$hash|" -e "s|${mktplcRef.version}|$version|" pkgs/applications/editors/vscode/extensions/python/default.nix
+    fi
+  '';
+
   meta = with lib; {
+    description = "A Visual Studio Code extension with rich support for the Python language";
+    downloadPage = "https://marketplace.visualstudio.com/items?itemName=ms-python.python";
+    homepage = "https://github.com/Microsoft/vscode-python";
+    changelog = "https://github.com/microsoft/vscode-python/releases";
```
[the remark this answers is not in the document]
Comment (superherointj) [reply pkgs/applications/editors/vscode/extensions/python/default.nix:81]: https://github.com/microsoft/vscode-python/blob/main/CHANGELOG.md
(Then you´d have to use the tag version. Same as being downloaded.)

[the remark this answers is not in the document]
Comment (linsui) [reply pkgs/applications/editors/vscode/extensions/python/default.nix:81]: But we don't have a 2022.14.0 version. :shrug:

[the remark this answers is not in the document]
Comment (linsui) [reply pkgs/applications/editors/vscode/extensions/python/default.nix:81]: So I should remove the changelog or?

[the remark this answers is not in the document]
Comment (superherointj) [reply pkgs/applications/editors/vscode/extensions/python/default.nix:81]: No, nevermind. I have already merged as it is.

**Comment by linsui** [comment]
Hi, @ryantm Do you know why this is not updated? Thanks!

**Pull request merged**